### PR TITLE
Add "optimisable" processed property

### PR DESCRIPTION
**Pull request by jackirvine97**
This pull request enables a user to flag a processed property as `optimisable`. This enables the following use cases:

1. The user supplies a single value - `pyproprop` returns said value
2. The user supplies a valid* set of bounds - `pyproprop` returns said bounds as a `tuple`
3. The user supplies a set of equal bounds -  `pyproprop` returns one of the bounds as a single value.

*Upper bound exceeds lower bound

### CODE REVIEW

File: pyproprop/processed_property.py [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 """
 
+from numbers import Real
+import numpy as np
```
Comment (brocksam) [inline pyproprop/processed_property.py:11]: Numpy is third party not standard library, import should be separated below

File: requirements.txt [hunk 2]
```diff
@@ -1 +1,2 @@
 numpy>=1.17
+typing>=3.7
```
Comment (brocksam) [inline requirements.txt:2]: Is typing not standard library? Does it need to be in requirements as it should come bundled with Python? Please correct me if I'm wrong

Comment (brocksam) [reply requirements.txt:2]: Ignore this I was wrong

**Comment by brocksam** [comment]
Struggling to understand why the Appveyor build failed...

File: pyproprop/processed_property.py (hunk 1 above)
Comment (jackirvine97) [reply pyproprop/processed_property.py]: Addressed in commit 5f733d9cdce2f00a55fa62ce381a13f017a3a960

**Comment by brocksam** [comment]
> Struggling to understand why the Appveyor build failed...

Actually seems to be a problem in the merging branch `optimisable` which doesn't contain the `appveyor.yml` file in `master` so will be resolved on PR merge.

**Comment by jackirvine97** [comment]
> Struggling to understand why the Appveyor build failed...

It failed in branch but passed in the PR. I wonder is it because the branch does not contain the most recent commits (eg 31228f285ad82f4142089790fcda31b30849a788) relating to appveyor, and when it is merged in with these recent commits it will pass.

**Pull request merged**